staging/src/k8s.io/cloud-provider/controllers/service/controller_test.go

```diff
@@ -910,6 +910,7 @@ func TestNodesNotEqual(t *testing.T) {
 	node3 := makeNode(tweakName("node3"))
 	node1WithProviderID := makeNode(tweakName("node1"), tweakProviderID("cumulus/1"))
 	node2WithProviderID := makeNode(tweakName("node2"), tweakProviderID("cumulus/2"))
+	node1WithUID := makeNode(tweakName("node1"), tweakUID("uid1"))
```

Comment:
hmm, all nodes have to have an UID, I think you should modify the `makeNode()` func to add a node UID automatically with `uuid.New(` per example

Reply:
I have tried to do this, but a few test fail. I don't want to change them in this small PR

Reply:
hmm, that is important, if the tests fails means there may be doing some wrong assumptions, or we are doing those, can you please submit the PR to see the failures?